lib/repl.js

```diff
@@ -1609,6 +1617,37 @@ function complete(line, callback) {
   }
 }
 
+async function includesProxiesOrGetters(fullExpr, evalFn, context) {
+  const bits = StringPrototypeSplit(fullExpr, '.');
+
+  let currentExpr = '';
+  for (let i = 0; i < bits.length - 1; i++) {
+    currentExpr += `${i === 0 ? '' : '.'}${bits[i]}`;
+    const currentExprIsObject = await evalPromisified(`try { ${currentExpr} !== null && typeof ${currentExpr} === 'object' } catch { false }`);
+    if (!currentExprIsObject) {
+      return false;
+    }
+
+    const currentExprIsProxy = await evalPromisified(`require("node:util/types").isProxy(${currentExpr})`);
+    if (currentExprIsProxy) {
+      return true;
+    }
+
+    const typeOfNextBitGet = await evalPromisified(`typeof Object.getOwnPropertyDescriptor(${currentExpr}, '${bits[i + 1]}')?.get`);
+    const nextBitHasGetter = typeOfNextBitGet === 'function';
+    if (nextBitHasGetter) {
+      return true;
+    }
+  }
+
+  function evalPromisified(evalExpr) {
+    return new Promise((resolve, reject) =>
```

Comment:
This can probably be replaced with `Promise.try(...)` but instead of using `new Promise(...)` but that would likely make it much more difficult to backport so I'll leave that up to you. Maybe worth adding a comment tho highlighting that that change could be made once it can be safely backported to older versions.

Reply:
Sorry I am not really familiar with `Promise.try` 😓

From [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/try) it sounds like you can call `Promise.try` on a callback, and it will be practically promisified right?

But here we are calling `evalFn` which takes a callback where we want to resolve from, so I am failing to see how `Promise.try` can help here, could you clarify what you have in mind? 🙏

Reply:
ah ok, thanks 😄👍